plugins/decklink/decklink-device-instance.cpp

```diff
@@ -288,6 +288,99 @@ bool DeckLinkDeviceInstance::StopCapture(void)
 	return true;
 }
 
+bool DeckLinkDeviceInstance::StartOutput(DeckLinkDeviceMode *mode_)
+{
+	if (mode != nullptr)
+		return false;
+	if (mode_ == nullptr)
+		return false;
+
+	LOG(LOG_INFO, "Starting output...");
+
+	if (!device->GetOutput(&output))
+		return false;
+
+	const HRESULT videoResult = output->EnableVideoOutput(mode_->GetDisplayMode(),
+										bmdVideoOutputFlagDefault);
+	if (videoResult != S_OK) {
+		LOG(LOG_ERROR, "Failed to enable video output");
+		return false;
+	}
+
+	const HRESULT audioResult = output->EnableAudioOutput(bmdAudioSampleRate48kHz, 
+										bmdAudioSampleType16bitInteger,
+										2,
+										bmdAudioOutputStreamTimestamped);
+	if (audioResult != S_OK) {
+		LOG(LOG_ERROR, "Failed to enable audio output");
+		return false;
+	}
+
+	mode = mode_;
+
+	auto decklinkOutput = dynamic_cast<DeckLinkOutput*>(decklink);
+	if (decklinkOutput == nullptr) {
+		return false;
+	}
+
+	HRESULT result;
+	result = output->CreateVideoFrame(decklinkOutput->GetWidth(), decklinkOutput->GetHeight(),
+					decklinkOutput->GetWidth()*2,
+					bmdFormat8BitYUV, bmdFrameFlagDefault, &decklinkOutputFrame);
+	if (result != S_OK) {
+		blog(LOG_ERROR ,"failed to make frame 0x%X", result);
+		return false;
+	}
+
+	return true;
+}
+
+bool DeckLinkDeviceInstance::StopOutput()
+{
+	if (mode == nullptr || output == nullptr)
+		return false;
+
+	LOG(LOG_INFO, "Stopping output of '%s'...",
+		GetDevice()->GetDisplayName().c_str());
+
+
+	output->DisableVideoOutput();
+
+	output->DisableAudioOutput();
+
+	if (decklinkOutputFrame != nullptr) {
+		decklinkOutputFrame->Release();
+		decklinkOutputFrame = nullptr;
+	}
+	
+	return true;
+}
+
+void DeckLinkDeviceInstance::DisplayVideoFrame(video_data *frame)
+{
+	auto decklinkOutput = dynamic_cast<DeckLinkOutput*>(decklink);
+	if (decklinkOutput == nullptr)
+		return;
+
+	uint8_t *destData;
+	decklinkOutputFrame->GetBytes((void**)&destData);
+
+	uint8_t *outData = frame->data[0];
+
+	memcpy(destData, outData,
+			(decklinkOutput->GetWidth() * decklinkOutput->GetHeight()) * 2);
```

Comment:
Style:  This needs line-broken differently.  Multiplication order doesn't matter, so you can probably lose the grouping parentheses.